### PR TITLE
InstructLab community knowledge

### PR DESCRIPTION
**Pull request by makelinux**
### 1: What is the main goal of the InstructLab project?
Pre-trainted:
_The InstructLab project aims to develop and optimize the production of a new generation of advanced materials for the pharmaceutical industry, specifically with the focus on developing a more sustainable production process for these materials. By utilizing a combination of computational simulations ..._

Provided:
The main goal of the InstructLab project is to leverage innovative techniques for overcoming challenges in LLM training, using taxonomy-based curation and synthetic data generation to enable contributions to existing LLMs.

Fine-tuned:
The InstructLab project aims to democratize contributions to AI and LLMs, allowing rapid model development through community collaboration facilitated by weekly builds that integrate community contributions. Additionally, it seeks to enable users to interact with and train the Merlinite-7b AI LLM by contributing skills and knowledge. The project also intends to create a more diverse set of training data for the LLM through community involvement. Ultimately, the goal is to build a stronger, more capable AI model through collective effort and continuous integration of community contributions. The project is currently focused on improving the skills and knowledge of the Merlinite-7b AI LLM through community-driven training data. Users can contribute new training data by following the [contribution guide](https://github.com/instructlab/instructlab/blob/main/CONTRIBUTING.md). The weekly builds integrate community contributions, allowing for continuous improvement of the LLM. The project also has an active [Slack community](https://join.slack.com/t/instruct-lab/shared_invite/zt-2ginke0oz-SxvNYZCqBTDy5wzAi9WRlA) 

### 2: How can users contribute to the InstructLab project?
Pre-trainted:
_InstructLab is an open-source project, which means that users can contribute in various ways to help improve and expand its capabilities. Here are some ways users can contribute:

1. **Reporting Bugs:** If users encounter any issues or bugs while using InstructLab, they can report them through the p..._

Provided:
Users can contribute to the InstructLab project by submitting new features, resolving bugs and issues, and refining documentation through pull requests. They can also contribute their own information and skills to the project's taxonomy repository on GitHub.

Fine-tuned:
Users can contribute to the InstructLab project in several ways, including:
- Submitting bug reports, feature requests, or pull requests through the project's GitHub repository.
- Participating in discussions on the project's mailing lists, Slack channels, or issue tracker.
- Sharing their experiences, insights, or feedback with the InstructLab community.
- Providing translation, documentation, or other non-coding contributions.
- Attending or presenting at InstructLab-related events, workshops, or meetups.
- Staying informed about the project's progress, direction, or announcements.
By engaging in these activities, users can help improve the InstructLab project, build relationships with other contributors, and expand their own skills and knowledge. Users should always follow the project's contribution guidelines, code of conduct, and other relevant policies to ensure their efforts are aligned with the project's goals and values. Users should also be respectful, courteous, and considerate in their interactions with other members of the InstructLab community, and should avoid engaging in harmful or inappropriate behavior. Users should also be transparent, honest, and responsible in their 

### 3: What is the purpose of the ilab CLI tool?
Pre-trainted:
_The ilab CLI tool is a command-line interface for managing and interacting with IBM Cloud Private clusters. It allows you to perform various operations, such as creating, deleting, and managing cluster resources, as well as executing custom commands on your cluster nodes. The ilab CLI tool is design..._

Provided:
The ilab CLI tool allows users to interact with LLMs, such as IBM AI models Merlinite or Granite. It facilitates augmenting the capabilities of these models by enabling users to contribute and train them locally.

Fine-tuned:
The ilab CLI tool is designed for interacting with the IBM `Lab Services` from the command line, allowing users to create, manage, and deploy their models more efficiently. It provides a range of subcommands for various tasks, such as logging in, creating a new model, listing available services, and deploying the model to a running IBM `Lab Services` instance. By using the ilab CLI tool, users can streamline their workflow and focus on developing their models rather than dealing with complex command-line commands. The ilab CLI tool also supports autocompletion for shells like Bash, Zsh, and Fish, making it easier for users to discover and use the available subcommands. Overall, the ilab CLI tool is a powerful and convenient tool for interacting with the IBM `Lab Services` from the command line. 

### 4: How can users see the impact of their contributions on an AI model?
Pre-trainted:
_Users can see the impact of their contributions on an AI model by analyzing the model's performance and comparing it with and without the user's contributions. This can be done by evaluating the model's accuracy, precision, recall, and other relevant metrics. Additionally, users can also check if th..._

Provided:
After contributing files to the main AI model through the taxonomy repository and having their pull request merged, users can see the impact of their contributions when the Merlinite or Granite models are updated. They can observe locally that the model has improved with the skills or knowledge they taught it.

Fine-tuned:
Users can see the impact of their contributions on an AI model by analyzing the model's performance or behavior before and after the contribution. If the contribution has had a significant effect, the model's output should change noticeably. Additionally, the project team may provide tools or metrics for measuring the impact of contributions specifically. If users are unsure whether their contribution has made a difference, they can always reach out to the project team for guidance. The team is usually happy to help and provide feedback on contributions, making it easier for users to understand the impact of their work. Remember, contributing to open source projects is a great way to learn and develop skills, even if the contribution seems small or insignificant. Every little bit helps, and the journey of a thousand miles begins with a single step! So, don't be afraid to dive in and give it a try. Happy coding! 💻🔧🌟 

### 5: What is the Developer Certificate of Origin (DCO) and why is it important for contributors?
Pre-trainted:
_The Developer Certificate of Origin (DCO) is a statement that contributors to open-source projects sign to verify that they have the right to submit their contributions and that they understand the project's licensing requirements. The DCO is an alternative to a traditional signed-off-by (SoB) commi..._

Provided:
The Developer Certificate of Origin (DCO) is a lightweight way for contributors to certify that they have the right to submit the code they are contributing to the project. It ensures that contributions adhere to the project's open source license and legal requirements. Contributors must sign off on their commits by adding a "Signed-off-by" line to certify their adherence to the DCO.

Fine-tuned:
The Developer Certificate of Origin (DCO) is a signature procedure that requires contributors to commit their changes using a valid email address. This ensures that the contributor can be identified and holds responsibility for their contributions. The DCO is important because it helps maintain the integrity of the project, makes it easier for developers to track down the original author of a change, and allows for automated build systems to enforce the requirement. The DCO also serves as a reminder that contributors are agreeing to follow the project's code of conduct and licensing terms. By using the DCO, contributors demonstrate their commitment to the project and show that they understand and accept the associated responsibilities. The DCO is particularly important for open source projects, where maintaining trust and accountability is crucial for long-term success. Ultimately, the DCO helps ensure that the project remains transparent, inclusive, and well-maintained, making it easier for developers to contribute and collaborate effectively. By using the DCO, contributors can help build a strong, sustainable community around the project, ensuring its long-term success and relevance in the rapidly evolving world of technology. Therefore, it is essential for contributors to understand and adhere to the DCO, as it plays a critical 

---

**Contribution checklist**

- [x] The contribution was tested with `ilab generate`
- [x] No errors or warnings were produced by `ilab generate`
- [x] All [commits are signed off](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [x] The `qna.yaml` file contains at least 5 `seed_examples`
- [x] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)
- [x] An `attribution.txt` file in the same folder as the `qna.yaml` file
- [x] Content does not include PII or otherwise sensitive or confidential information
- [x] Content does not include anything documented in the project's [Avoid these Topics](https://github.com/instructlab/taxonomy/blob/main/docs/SKILLS_GUIDE.md#avoid-these-topics) guidelines

### CODE REVIEW

**Comment by jjasghar** [comment]
Can you fix the linting issue please?

File: knowledge/technical_manual/instructlab/attribution.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+https://github.com/instructlab/
```
Comment (bjhargrave) [inline knowledge/technical_manual/instructlab/attribution.txt:1]: This is not a proper attribution file. Please see the readme for the expected content.

File: knowledge/technical_manual/instructlab/qna.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+created_by: makelinux
+domain: AI
+seed_examples:
+  - question: What is InstructLab?
+    answer: InstructLab (Large-scale Alignment for chatBots) is an open source
+      initiative by Red Hat and IBM. It provides a platform for easy engagement
+      with LLM.
+  - question: How users can use InstructLab?
```
Comment (bjhargrave) [inline knowledge/technical_manual/instructlab/qna.yaml:9]: I think the questions and answers need improvement to be better and more complete english sentences.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: knowledge/technical_manual/instructlab/qna.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+created_by: makelinux
+domain: AI
+seed_examples:
+  - question: What is the main goal of the InstructLab project?
+    answer: The main goal of the InstructLab project is to leverage innovative
+      techniques for overcoming challenges in LLM training, using
+      taxonomy-based curation and synthetic data generation to enable
+      contributions to existing LLMs.
```
Comment (bjhargrave) [inline knowledge/technical_manual/instructlab/qna.yaml:9]:
```suggestion
    answer: |
      The main goal of the InstructLab project is to leverage innovative
      techniques for overcoming challenges in LLM training, using
      taxonomy-based curation and synthetic data generation to enable
      contributions to existing LLMs.
```

See https://yaml-multiline.info/ for info on using multi-line strings in yaml.

Your other answers will need similar fixes.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: knowledge/technical_manual/instructlab/qna.yaml (hunk 3 above)
Comment (jjasghar) [reply knowledge/technical_manual/instructlab/qna.yaml:9]: Oh, BJ, that is a great link. I'm stealing it.

Comment (makelinux) [reply knowledge/technical_manual/instructlab/qna.yaml:9]: @bjhargrave , why do suggest "literal style, indicated by a pipe (|)." and not "folded style, indicated by a right angle bracket (>). "?

Comment (bjhargrave) [reply knowledge/technical_manual/instructlab/qna.yaml:9]: It retains the new lines which can be important for certain information like tables. So we generally suggest `|` as the default choice here. If your value is pure text where new lines are just whitespace, `>` is fine also.

Comment (bjhargrave) [reply knowledge/technical_manual/instructlab/qna.yaml:9]: What you have is ok as a plain flow scalar.

Comment (makelinux) [reply knowledge/technical_manual/instructlab/qna.yaml:9]: @bjhargrave , so leave it as it is or add `>-`? 

> To remove all newlines, strip them by putting a minus sign (-)

**Comment by bjhargrave** [comment]
@instructlab-bot precheck

**Comment by bjhargrave** [comment]
@instructlab-bot generate

**Review by bjhargrave — APPROVED** [review]

**Comment by bjhargrave** [comment]
The SDG looks very good. Approve for next training run.

**Review by jjasghar — APPROVED** [review]
This looks great, thank you for this!

**Comment by makelinux** [comment]
What are obstacles to be approved?

**Comment by jjasghar** [comment]
We are waiting for our end-to-end run to be completed. It will be picked up in the next one, and we'll see how it changes the model upstream.
Unfortunately I can't tell you _when_ it will be, but it is marked as "triage-approved," so it will be picked up.

**Comment by jjasghar** [comment]
Hi! :wave:
With the new update and release of 0.18.0, our [base knowledge submissions](https://github.com/instructlab/taxonomy/?tab=readme-ov-file#knowledge-yaml-examples) have drastically changed. This has given us an opportunity to significantly improve the accuracy of our fine-tuned models, which is a major win for our community!

But alas, that has caused some changes that this PR isn't able to continue forward. We greatly appreciate the work you've put into this, and wish we could accept it but the amount of work to get it to be `version: 3` is unrealistic for us to ask.
If you _do_ want to do this, please don't hesitate to take the `qna.yaml` update it, and build it out. Otherwise, we hope you come back to our community with a new submission and help build out our Open Source community-driven LLM.

If you have any questions directly, please don't hesitate to comment on the PR or email me at [awesomeATinstructlab.ai](http://awesomeatinstructlab.ai/) (yes, really), and I'll do my best to answer your questions.
